skimage/future/graph/rag.py

```diff
@@ -157,6 +157,21 @@ def __init__(self, label_image=None, connectivity=1, data=None, **attr):
                                   strides=((0,) * label_image.ndim)),
                 extra_arguments=(self,))
 
+    def __eq__(self, other):
```

Comment:
Please make changes to this - I am not very familiar with this class, but added all of the things I could think of while investigating this issue.

Reply:
Maybe just move this function to the test methods. It seems that would avoid the introduction of a new public method. You can add a comment that you think it might be useful to move it back to the class in the future

Reply:
My preference is for separate PRs, this method needs to be modified to consider edges, as @jni pointed out.